redhat/frr.spec.in

```diff
@@ -155,14 +155,12 @@ BuildRequires:  python27-devel python27-sphinx
 %else
 BuildRequires:  python-devel >= 2.7 python-sphinx
 %endif
-Requires:       ncurses json-c initscripts
+Requires:       json-c initscripts
 %if %{with_pam}
 BuildRequires:  pam-devel
-Requires:       pam
```

Comment:
Are you sure with this? `BuildRequires` should be requirement to build the RPM, while `Requires` is for installing the (binary) RPM.
`pam-devel` is needed for building (and will include `pam`, so no need to specify it for `BuildRequires`), but we still need to make sure to have `pam`required for the binary package

Reply:
Yes, rpm handles this just fine. It basically runs ldd on the frr binary and creates Requires for the linked shared libs. You can verify this with rpm -q --requires frr.
http://ftp.rpm.org/max-rpm/s1-rpm-depend-auto-depend.html describes it in more detail. It's old but still valid.

Reply:
And still more info here: https://fedoraproject.org/wiki/Packaging:Guidelines#Explicit_Requires

Reply:
Thanks for explanation. Needed to test this myself on some older systems (CentOS 6), but seems to work as you described